text/0000-confidential-token-ids.md

```diff
@@ -0,0 +1,270 @@
+
+- To prove each TXO's commitment to the `amount` value, we use a Pedersen commitment, consisting of a group generator (`H_i`) raised to the value of the `amount`, with a blinding factor to prevent brute-forcing the full range of values (`2^64`). We compute the blinding factor as the blinding group generator raised to a hash of the shared secret for the TXO.
+
+    ```
+    blinding_factor = Blake2B(blinding_tag | shared_secret)
+    commitment = H_i * Scalar::from(amount) + G * blinding_factor
+    ```
+
+    - Note, the group generators for Token Types are computed using hashing to a curve so that they are "orthogonal" for computable adversaries, meaning there is no discernable, nontrivial linear relationship between base points. See our [transaction core group generators reference](https://github.com/mobilecoinfoundation/mobilecoin/blob/master/transaction/core/src/ring_signature/mod.rs#L29). 
+    - Note also, the blinding factor base point does not need to be unique per TokenType, so we use the same blinding factor base point for all TokenTypes.
+    - Note finally, the Proof of Opening requires all commitments in a transaction were generated using the same group generator, base point `H_i`. If the commitments in input and/or output TXOs in a transaction are constructed with different group generators, and the sender is able to unwrap the sum as specified in the protocol below, it implies that the sender knows a nontrivial linear relationship among the `H_i`, which contradicts the assumption that the `H_i` are orthogonal. 
+
+The non-interactive zero knowledge Proof of Opening protocol is the following: (See [Zero Knowledge Proofs and Commitment Schemes, Page 27](https://www.cs.purdue.edu/homes/ninghui/courses/555_Spring12/handouts/555_Spring12_topic23.pdf))
+
+
+1. The prover establishes a value, `d`, calculated by hashing their secret values from the commitment (the `token_id` and the `blinding_factor`). Implementation note, we may use [Merlin](https://docs.rs/merlin/latest/merlin/) rather than Blake2B.
+
+    ```
+    y = Blake2B(proof_tag_t | token_id)
+    s = Blake2B(proof_tag_b | blinding_factor)
+    d = G * y + H * s
+    ```
+
+2. The non-interactive verifier chooses challenge, `e`, by hashing the commitment and the value `d` from the previous step.
+
+    ```
+    e = Blake2B(proof_tag_e | commitment | d)
+    ```
+
+3. The prover computes the following, and includes these values in the Proof of Opening:
+
+    ```
+    u = y + e * token_id
```

Comment:
Is this supposed to be `u = y + e * amount`?

Don't you need a range proof on `commitment`? Am I wrong in thinking range proofs qualify as 'proofs of opening'?

Reply:
You were correct! And we also removed the Proof of Opening from the proposal